test/models/concerns/feed_ad_zone_test.rb

```diff
@@ -0,0 +1,118 @@
+require "test_helper"
+
+class FeedAdZoneTest < ActiveSupport::TestCase
+  let(:podcast) { create(:podcast) }
+  let(:feed) { podcast.default_feed }
+
+  describe "getter methods" do
+    before do
+      assert_nil feed.include_zones
+    end
+
+    describe "#billboard" do
+      it "returns include_zones check value" do
+        assert_equal feed.billboard, "1"
+        feed.include_zones = []
+        assert_equal feed.billboard, "0"
+        feed.include_zones << "billboard"
+        assert_equal feed.billboard, "1"
+      end
+    end
+
+    describe "#house" do
+      it "returns include_zones check value" do
+        assert_equal feed.house, "1"
+        feed.include_zones = []
+        assert_equal feed.house, "0"
+        feed.include_zones << "house"
+        assert_equal feed.house, "1"
+      end
+    end
+
+    describe "#paid" do
+      it "returns include_zones check value" do
+        assert_equal feed.paid, "1"
+        feed.include_zones = []
+        assert_equal feed.paid, "0"
+        feed.include_zones << "ad"
+        assert_equal feed.paid, "1"
+      end
+    end
+
+    describe "#sonic_id" do
+      it "returns include_zones check value" do
+        assert_equal feed.sonic_id, "1"
+        feed.include_zones = []
+        assert_equal feed.sonic_id, "0"
+        feed.include_zones << "sonic_id"
+        assert_equal feed.sonic_id, "1"
+      end
+    end
+  end
+
+  describe "setter methods" do
+    let(:all_zones) { ["billboard", "house", "ad", "sonic_id"] }
+
+    describe "#add_zone" do
```

Comment:
this suite and `#remove_zone` are possibly misnomers? since i don't test either of those methods directly (they're both private), but rather through calling each of the `method=` setters. so i'm not sure if i should have gone the more redundant route and just had each of those be their own suites and repeating the same tests in each? thinking about long-term readability if this is the kind of thing that would be confusing to understand a year from now.

Reply:
Ah, I see what you're saying.

I think either way is fine - though it is a bit odd to reference a private method here, even as a comment.  Stylistically, I might prefer the more verbose style?  Or write one mega-test, and just loop over all 4 zone names, running the same checks?

Your call.  Definitely multiple ways.

Reply:
ended up rewriting the verbose way and i think it reads better.